### PR TITLE
docs: Add warnings about using layer 7 LBs with TLS routing

### PR DESCRIPTION
**Pull request by webvictim**
It's become increasingly obvious from the community Slack and discussion posts recently that many people are attempting to deploy Teleport in its default TLS routing mode behind HTTP reverse proxies like nginx or HAProxy and are wondering why it won't work. I'm hoping that adding an explicit section in the docs (which will show up if you search for nginx, ALB, haproxy, "failed reading version string" or similar) should try to help them avoid this all-too-common footgun.

Also removed an old note saying that Teleport Cloud does not support TLS routing, as it now does.

There is probably a lot more scope to make this obvious in the docs, as well as value in adding guides which detail exactly how to configure Teleport behind a reverse proxy.

### CODE REVIEW

File: docs/pages/architecture/tls-routing.mdx [hunk 1]
```diff
@@ -25,6 +25,33 @@ To implement TLS routing, Teleport uses SNI ([Server Name Indication](https://en
...
+Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport web UI experience that seems
+to work perfectly, but the use of `tsh`, `tctl` and attempting to join remote Teleport services to the cluster will fail
+with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport web UI is not always an indication
+of a correctly configured Teleport cluster.
+
+If in doubt, remove all load balancers/proxies from the equation and connect to Teleport's web port directly to isolate the issue.
+
+To use Teleport behind a reverse proxy, you should either:
+- use a layer 4 (TCP) proxy which forwards TCP streams directly to Teleport (which will in turn handle TLS termination itself)
+- deploy Teleport in its [separate ports](../reference/networking.mdx#ports-without-tls-routing) mode by adding `version: v1` to `/etc/teleport.yaml` and removing `proxy_listener_mode: multiplex`
```
Comment (zmb3) [inline docs/pages/architecture/tls-routing.mdx:52]: Do we ever refer to "separate ports mode" anywhere else? My guess is no.

I would rephrase this to say: "disable Teleport's TLS routing by adding..."

**Review by zmb3 — COMMENTED** [review]
Would a link to this section from `docs/pages/management/admin/troubleshooting.mdx` also make sense?

**Comment by alexfornuto** [comment]
> Would a link to this section from `docs/pages/management/admin/troubleshooting.mdx` also make sense?

We could also make this new content a partial file and spread it around. The include directive could be wrapped in a Details block in places where want to include the information but not increase the default page size too much.

Not sure if this makes sense or where else we would include it, just an idea.

**Review by alexfornuto — APPROVED** [review]
Copy looks good, pending the comments above.

File: docs/pages/architecture/tls-routing.mdx (hunk 1 above)
Comment (strideynet) [inline docs/pages/architecture/tls-routing.mdx:52]: It's a bit of a random musing - but - if this is a supported configuration which we are recommending for environments with a L7 LB, we really need to introduce a way to put Teleport into "ports separate" mode that isn't "set your config to V1" as this is going to become more and more problematic as we start introducing future configuration versions. What if I want port separate mode and separate configuration of proxy and auth server addresses - I'll see if we have an outstanding issue for this and raise it.

Comment (webvictim) [reply docs/pages/architecture/tls-routing.mdx]: Being totally honest, I hated that we made TLS routing the default mode for exactly this reason - at least 50% of our AWS customers are using ACM and many other customers use reverse proxies. I would personally say we always need to maintain, support and test a non-TLS routing way of deploying Teleport. I would love to see better support for this.

**Comment by webvictim** [comment]
I've updated the info to an include and added it to both mentioned pages. It is a lot of content, but IMO it's important that we surface the information in an obvious way.

We can maybe look to add this to more places in future in a more condensed/default closed format.

File: docs/pages/architecture/tls-routing.mdx [hunk 2]
```diff
@@ -25,6 +25,13 @@ To implement TLS routing, Teleport uses SNI ([Server Name Indication](https://en
 and ALPN ([Application-Level Protocol Negotiation](https://en.wikipedia.org/wiki/Application-Layer_Protocol_Negotiation))
 TLS extensions.
 
+<Admonition
```
Comment (ptgott) [inline docs/pages/architecture/tls-routing.mdx:28]: Since this is Admonition is rather long, I think it would make sense to turn this into an H2-level section, e.g., "## When to use TLS routing", with the same content.

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Using Teleport's TLS routing mode behind a layer 7 (HTTP/HTTPS) proxy is generally not supported, due to
+these proxies terminating TLS themselves and then rewriting their requests to the upstream service, stripping
+the additional SNI/ALPN parts of the request in the process. In order for ALPN to work correctly, the Teleport
+proxy must terminate TLS itself.
+
+Broadly, this means that Teleport's TLS routing functionality is incompatible with:
+- AWS ALBs (Application Load Balancers)
+- AWS NLBs (Network Load Balancers), when using a TLS listener and a public ACM (Amazon Certificate Manager) certificate
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:8]:
```suggestion
- AWS NLBs (Network Load Balancers) when using a TLS listener and a public ACM (Amazon Certificate Manager) certificate
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+- Commonly used HTTP reverse proxies including nginx, Apache, Caddy, Traefik, HAProxy and many others
+- Cloudflare tunnels in their default configuration
+
+Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport web UI experience that seems
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:12]:
```suggestion
Deploying Teleport in TLS routing mode behind an HTTP proxy will result in a Teleport Web UI experience that seems
```
To be consistent with the use of this as a product name in the docs

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+to work perfectly, but the use of `tsh`, `tctl` and attempting to join remote Teleport services to the cluster will fail
+with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport web UI is not always an indication
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:14]:
```suggestion
with errors like `ssh: overflow reading version string` and `EOF`. A functioning Teleport Web UI is not always an indication
```

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+of a correctly configured Teleport cluster.
+
+If in doubt, remove all load balancers/proxies from the equation and connect to Teleport's web port directly to isolate the issue.
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:17]: Does "connect" refer to client connections or connecting Teleport services? Does "Teleport" refer to the Proxy Service?

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+To use Teleport behind a reverse proxy, you should either:
+- use a layer 4 (TCP) proxy which forwards TCP streams directly to Teleport (which will in turn handle TLS termination itself)
+- disable Teleport's TLS routing mode by adding `version: v1` to your config file and removing `proxy_listener_mode: multiplex`
+  - You can get an example v1 config file using `teleport configure --version=v1 --public-addr=teleport.example.com:443` (change the public address to your own domain)
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:22]:
```suggestion
 
You can get an example v1 config file using `teleport configure --version=v1 --public-addr=teleport.example.com:443` (change the public address to your own domain)
```
Since this is a new thought rather than a continuation of the list (it doesn't follow "you should either:")

File: docs/pages/includes/tls-multiplexing-warnings.mdx [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  - A list of ports which will be used can be found here: [ports without TLS routing](../reference/networking.mdx#ports-without-tls-routing)
```
Comment (ptgott) [inline docs/pages/includes/tls-multiplexing-warnings.mdx:23]:
```suggestion
If disabling TLS routing, you can find a list of ports to use for connecting different Teleport services at [ports without TLS routing](../reference/networking.mdx#ports-without-tls-routing)
```
To make this a bit more specific

File: docs/pages/includes/tls-multiplexing-warnings.mdx (hunk 6 above)
Comment (webvictim) [reply docs/pages/includes/tls-multiplexing-warnings.mdx:17]: Both. I'll clarify.

File: docs/pages/architecture/tls-routing.mdx (hunk 2 above)
Comment (webvictim) [reply docs/pages/architecture/tls-routing.mdx:28]: I like the idea of using a warning Admonition because it puts a big yellow box on the page and makes look at it. I'll move the content to a separate section and trim down the Admonition to hint that TLS routing may not be for everyone.

**Comment by alexfornuto** [comment]
Don't forget the appropriate backport labels 😄

**Review by ptgott — APPROVED** [review]
